### ISSUE TITLE
+1 miscellaneous sound fix - for chaingunners not to emit shotgun sounds when firing?

### ISSUE THREAD

**Issue by Zodomaniac** [fabiangreffrath/crispy-doom#329]
How could such an obvious one stay out of scope for so long? :)
I searched and found this Doomworld thread with the explaining comment:
https://www.doomworld.com/forum/topic/51919-heavy-weapons-guy-uses-shotgun/?do=findComment&comment=894036

In fact there is a reason why the shotgun sound is used, and there's even a DeHackEd patch for those who prefer the proper sound, changing the chaingunners' behavior though:

> The spiderdemon uses the shotgun sound because it's using a super chaingun that does much more damage. It shoots 3 pellets per shot, instead of one, just like the shotgun guy. We can assume it's shooting bigger bullets, more bullets or even shotgun shells.
> 
> The heavy weapon dude uses the shotgun sound because its shots last 4 tics, like the player's, but there's an extra "target checking" tic after every two shots, which makes their attack sounds a bit slower than the player's. At one point, the Eternity engine had made this change. Complaints ensued because the heavy weapon dudes sounded weaker than the player, and it was changed back to the shotgun sound. With the shotgun sound, their chaingun sounds beefy enough without altering game behavior (the speed of the monster's attack).
> 
> To be honest, I didn't notice they used the shotgun sound till long after I started playing DOOM II. To me, these days, such a difference is just another of the game's usual quirks and not a problem.
> 
> I've attached a little DeHackEd patch https://www.doomworld.com/applications/core/interface/file/attachment.php?id=1989 that gives the heavy weapon dude the attack sound and rate of the player. Note that it's incompatible with the usual game behavior, so it's not usable when watching recorded demos or when playing online with people not applying the patch. It makes the heavy weapon dudes a little bit more deadly than usual.

**Comment by Zodomaniac** [comment]
I tried it and discovered it doesn't work with Black Ops https://www.doomworld.com/forum/topic/84859-black-ops-smooth-weapons-dehacked-mod/ and Smooth Monsters https://www.doomworld.com/forum/topic/85991-smoothed-smooth-monsters-for-doom-retro-and-crispy-doom/ addons, could comeone tweak it?

**Comment by fabiangreffrath** [comment]
Well, the fact that the DEHACKED solution doesn't work well together with other mods doesn't mean it's a good idea to implement this in the port. The best reason not to do this so already given in the quoted text:

> At one point, the Eternity engine had made this change. Complaints ensued because the heavy weapon dudes sounded weaker than the player, and it was changed back to the shotgun sound.

**Comment by Zodomaniac** [comment]
OK, I think it would make sense to keep this open until someone makes a 'Crispy Edition' of the above patch that would work with enhancer mods, as I requested at Doomworld forums: https://www.doomworld.com/forum/topic/51919-heavy-weapons-guy-uses-shotgun/

**Comment by fabiangreffrath** [comment]
Nevertheless i'd be curious how this sounds.

```
--- a/src/doom/p_enemy.c
+++ b/src/doom/p_enemy.c
@@ -864,7 +864,7 @@ void A_CPosAttack (mobj_t* actor)
     if (!actor->target)
        return;

-    S_StartSound (actor, sfx_shotgn);
+    S_StartSound (actor, (actor->type == 65 && crispy->soundfix) ? sfx_pistol : sfx_shotgn);
     A_FaceTarget (actor);
     bangle = actor->angle;
     slope = P_AimLineAttack (actor, bangle, MISSILERANGE);

```

**Comment by fabiangreffrath** [comment]
I don't think it sounds too bad with this patch, though - on the other hand - the change might be considered quite severe and it doesn't feel like a revelation, either.

**Comment by Zodomaniac** [comment]
Here's the Smoothed Monsters add-on with my tweak for Chaingunner: https://www.doomworld.com/forum/topic/85991-smoothed-smooth-monsters-for-doom-retro-and-crispy-doom/?do=findComment&comment=1920849
